cosmoz-default-tree.js

```diff
@@ -103,14 +106,17 @@ export class DefaultTree extends Tree {
 	 * @param {Boolean} options.exact [false] (If the search should be executed exact or fuzzy. true wouldn't match "Pet")
 	 * @param {Boolean} options.firstHitOnly [false] (If the search should only return the first found node.)
 	 * @param {Object} nodes [this._roots] (The nodes the search should be based on.)
-	*/
+	 */
 	_searchNodes(propertyValue, options, nodes = this._roots) {
-		let results = [];
+		const results = [];
 
-		nodes.some(node => {
-			results = results.concat(this.search(node, propertyValue, options));
-			return options.firstHitOnly && results.length > 0;
-		});
+		for (const node of nodes) {
```

Comment:
We mitigate some performance & possible memory heap increases by not createing (and discarding) an extra array at each loop step.

Reply:
if firstHitOnly is true, will not results array be empty? since we return res before it can push the res?

Reply:
If firstHitOnly is true and if `res` (temporary results) are not empty than we return them. 
Same as before.

Reply:
Ah right, first hit, sorry not thinking.